In [1]:
use practice;

Commands completed successfully

Total execution time: 00:00:00

In [36]:
select * from baseball;

(823 row(s) affected)

Total execution time: 00:00:01.023

Num Name Team Games AtBats Runs Hits Doubles Triples Homers RBIs 2 José Abreu CHW 159 634 85 180 38 1 33 123 3 Ronald Acuna Jr. ATL 156 626 127 175 22 2 41 101 5 Cristhian Adames SFG 10 22 1 7 1 0 0 2 6 Willy Adames TBR 152 531 69 135 25 1 20 52 12 Matt Adams WSN 111 310 42 70 14 0 20 56 14 Ehire Adrianza MIN 83 202 34 55 8 3 5 22 15 Dario Agrazal PIT 14 22 0 2 0 0 0 2 17 Jesus Aguilar MIL 94 222 26 50 9 0 8 34 18 Jesus Aguilar TBR 37 92 13 24 3 0 4 16 19 Nick Ahmed ARI 158 556 79 141 33 6 19 82 21 R.J. Alaniz CIN 8 1 0 1 0 0 0 1 23 Hanser Alberto BAL 139 524 62 160 21 2 12 51 24 Ozzie Albies ATL 160 640 102 189 43 8 24 86 25 Sandy Alcantara MIA 31 58 2 5 1 0 0 4 28 Jorge Alfaro MIA 130 431 44 113 14 1 18 57 29 Anthony Alford TOR 16 28 3 5 0 0 1 1 31 Austin Allen SDP 34 65 4 14 4 0 0 3 33 Greg Allen CLE 89 231 30 53 9 3 4 27 35 Logan Allen SDP 7 4 1 1 0 0 0 0 36 Abraham Almonte ARI 17 31 11 9 3 1 1 4 38 Albert Almora CHC 130 339 41 80 11 1 12 32 39 Pete Alonso NYM 161 597 103 155 30 2 53 120 41 Yonder Alonso CHW 67 219 23 39 6 0 7 27 42 Yonder Alonso COL 54 73 11 19 7 0 3 10 45 Aaron Altherr PHI 22 29 2 1 1 0 0 1 47 Aaron Altherr NYM 26 31 6 4 1 0 1 2 48 Jose Altuve HOU 124 500 89 149 27 3 31 74 51 Yordan Álvarez HOU 87 313 58 98 26 0 27 78 53 Brett Anderson OAK 3 8 1 4 0 0 0 0 54 Brian Anderson MIA 126 459 57 120 33 1 20 66 55 Chase Anderson MIL 31 40 1 4 1 0 0 1 61 Shaun Anderson SFG 25 25 0 2 1 0 0 0 63 Tim Anderson CHW 123 498 81 167 32 0 18 56 64 Tyler Anderson COL 5 7 1 1 0 0 0 0 66 Elvis Andrus TEX 147 600 81 165 27 4 12 72 67 Miguel Andujar NYY 12 47 1 6 0 0 0 1 68 Aristides Aquino CIN 56 205 31 53 8 0 19 47 70 Chris Archer PIT 22 35 1 4 0 0 0 2 71 Orlando Arcia MIL 152 494 51 110 16 1 15 59 72 Nolan Arenado COL 155 588 102 185 31 2 41 118 75 Randy Arozarena STL 19 20 4 6 1 0 1 2 76 Luis Arraez MIN 92 326 54 109 20 1 4 28 77 Jake Arrieta PHI 23 45 2 7 1 0 0 4 78 Christian Arroyo TBR 16 50 8 11 2 0 2 7 79 Humberto Arteaga KCR 41 122 11 24 4 0 0 4 80 Willians Astudillo MIN 58 190 28 51 9 0 4 21 83 Tyler Austin MIN 2 4 1 1 1 0 0 0 84 Tyler Austin SFG 70 130 24 24 2 1 8 20 85 Tyler Austin MIL 17 20 5 4 2 0 1 4 86 Abiatal Avelino SFG 4 7 0 2 0 0 0 1 87 Alex Avila ARI 63 164 22 34 8 0 9 24 90 Harrison Bader STL 128 347 54 71 14 3 12 39 91 Javier Baez CHC 138 531 89 149 38 4 29 85 96 Homer Bailey OAK 2 4 0 2 0 0 0 1 100 Austin Barnes LAD 75 212 28 43 12 1 5 25 106 Tucker Barnhart CIN 114 316 32 73 14 0 11 40 107 Steven Baron PIT 7 10 0 2 1 0 0 1 112 Franklin Barreto OAK 23 57 6 7 2 0 2 5 119 Trevor Bauer CLE 2 5 0 1 0 0 0 0 120 Trevor Bauer CIN 9 14 1 2 0 0 0 0 121 Jake Bauers CLE 117 372 46 84 16 1 12 43 122 Matt Beaty LAD 99 249 36 66 19 1 9 46 123 Gordon Beckham DET 83 223 29 48 13 2 6 15 124 Tim Beckham SEA 88 304 39 72 21 1 15 47 127 Tyler Beede SFG 25 38 2 4 0 0 0 1 129 Josh Bell PIT 143 527 94 146 37 3 37 116 130 Cody Bellinger LAD 156 558 121 170 34 3 47 115 131 Brandon Belt SFG 156 526 76 123 32 3 17 57 133 Anthony Bemboom TBR 3 5 0 2 1 0 0 1 134 Anthony Bemboom LAA 22 49 2 5 0 0 1 3 135 Andrew Benintendi BOS 138 541 72 144 40 5 13 68 138 Jon Berti MIA 73 256 52 70 14 1 6 24 140 Mookie Betts BOS 150 597 135 176 40 5 29 80 144 Bo Bichette TOR 46 196 32 61 18 0 11 21 148 Cavan Biggio TOR 100 354 66 83 17 2 16 48 149 Greg Bird NYY 10 35 6 6 0 0 1 1 151 Braden Bishop SEA 27 56 3 6 0 0 0 4 157 Charlie Blackmon COL 140 580 112 182 42 7 32 86 158 Alex Blandino CIN 23 36 6 9 1 0 1 3 162 Xander Bogaerts BOS 155 614 110 190 52 0 33 117 164 Skye Bolt OAK 5 10 1 1 1 0 0 0 165 Jorge Bonifacio KCR 5 20 3 7 3 0 0 3 167 David Bote CHC 127 303 47 78 17 0 11 41 168 Justin Bour LAA 52 151 18 26 5 0 8 26 169 Peter Bourjos LAA 26 44 4 4 1 0 0 2 173 Matthew Boyd DET 1 2 0 1 0 0 0 0 175 Brad Brach CHC 39 4 0 1 0 0 0 1 179 Bobby Bradley CLE 15 45 4 8 5 0 1 4 180 Jackie Bradley Jr. BOS 147 494 69 111 28 3 21 62 181 Michael Brantley HOU 148 575 88 179 40 2 22 90 184 Steven Brault PIT 33 42 8 14 1 0 1 2 185 Ryan Braun MIL 144 459 70 131 31 2 2

1. **<u>Retrieve the total number of games played by each player.</u>**

In [37]:
SELECT Name, SUM(Games) AS TotalGames
FROM baseball
GROUP BY Name;


(768 row(s) affected)

Total execution time: 00:00:01.031

Name TotalGames José Abreu 159 Ronald Acuna Jr. 156 Cristhian Adames 10 Willy Adames 152 Matt Adams 111 Ehire Adrianza 83 Dario Agrazal 14 Jesus Aguilar 131 Nick Ahmed 158 R.J. Alaniz 8 Hanser Alberto 139 Ozzie Albies 160 Sandy Alcantara 31 Jorge Alfaro 130 Anthony Alford 16 Austin Allen 34 Greg Allen 89 Logan Allen 7 Abraham Almonte 17 Albert Almora 130 Pete Alonso 161 Yonder Alonso 121 Aaron Altherr 48 Jose Altuve 124 Yordan Álvarez 87 Brett Anderson 3 Brian Anderson 126 Chase Anderson 31 Shaun Anderson 25 Tim Anderson 123 Tyler Anderson 5 Elvis Andrus 147 Miguel Andujar 12 Aristides Aquino 56 Chris Archer 22 Orlando Arcia 152 Nolan Arenado 155 Randy Arozarena 19 Luis Arraez 92 Jake Arrieta 23 Christian Arroyo 16 Humberto Arteaga 41 Willians Astudillo 58 Tyler Austin 89 Abiatal Avelino 4 Alex Avila 63 Harrison Bader 128 Javier Baez 138 Homer Bailey 2 Austin Barnes 75 Tucker Barnhart 114 Steven Baron 7 Franklin Barreto 23 Trevor Bauer 11 Jake Bauers 117 Matt Beaty 99 Gordon Beckham 83 Tim Beckham 88 Tyler Beede 25 Josh Bell 143 Cody Bellinger 156 Brandon Belt 156 Anthony Bemboom 25 Andrew Benintendi 138 Jon Berti 73 Mookie Betts 150 Bo Bichette 46 Cavan Biggio 100 Greg Bird 10 Braden Bishop 27 Charlie Blackmon 140 Alex Blandino 23 Xander Bogaerts 155 Skye Bolt 5 Jorge Bonifacio 5 David Bote 127 Justin Bour 52 Peter Bourjos 26 Matthew Boyd 1 Brad Brach 39 Bobby Bradley 15 Jackie Bradley Jr. 147 Michael Brantley 148 Steven Brault 33 Ryan Braun 144 John Brebbia 63 Alex Bregman 156 Lewis Brinson 75 Socrates Brito 17 Mike Brosseau 50 Seth Brown 26 Keon Broxton 100 Jay Bruce 98 Kris Bryant 147 Walker Buehler 29 Madison Bumgarner 35 Corbin Burnes 30 Drew Butera 16 Byron Buxton 87 Asdrubal Cabrera 131 Melky Cabrera 133 Miguel Cabrera 136 Trevor Cahill 2 Lorenzo Cain 148 Kole Calhoun 152 Willie Calhoun 83 Johan Camargo 98 Jeimer Candelario 94 Mark Canha 126 Robinson Cano 107 Victor Caratini 95 Matt Carpenter 129 Curt Casali 84 Nicholas Castellanos 151 Luis Castillo 31 Welington Castillo 72 Wilkin Castillo 2 Harold Castro 97 Jason Castro 79 Starlin Castro 162 Willi Castro 30 Jake Cave 72 Juan Centeno 7 Francisco Cervelli 48 Jhoulys Chacin 18 Yu Chang 28 Matt Chapman 156 Tyler Chatwood 40 Jesse Chavez 4 Michael Chavis 95 Robinson Chirinos 114 Ji-Man Choi 127 Shin-Soo Choo 151 Nick Ciuffo 3 Taylor Clarke 21 Zack Collins 27 Christian Colon 8 Michael Conforto 151 Willson Contreras 105 Garrett Cooper 107 Patrick Corbin 31 Ryan Cordell 97 Franchy Cordero 9 Carlos Correa 75 Ryan Court 12 Kaleb Cowart 9 Zack Cozart 38 Brandon Crawford 147 J.P. Crawford 93 C.J. Cron 125 Kevin Cron 39 Nelson Cruz 120 Charlie Culberson 108 Cheslor Cuthbert 87 Travis d'Arnaud 102 David Dahl 100 Yu Darvish 30 Zach Davies 34 Chris Davis 105 J.D. Davis 140 Jaylin Davis 17 Johnny Davis 8 Jonathan Davis 37 Khris Davis 133 Rajai Davis 29 Taylor Davis 7 Yonathan Daza 44 Austin Dean 64 Jacob deGrom 31 Paul DeJong 159 Nicky Delmonico 21 Travis Demeritte 48 Daniel Descalso 82 Anthony DeSclafani 30 Delino DeShields 118 Ian Desmond 140 Rafael Devers 156 Aledmys Díaz 69 Elias Diaz 101 Isan Diaz 49 Yandy Díaz 79 Alex Dickerson 68 Corey Dickerson 78 Derek Dietrich 113 Wilmer Difo 43 Nick Dini 20 Brandon Dixon 117 Josh Donaldson 155 Brian Dozier 135 Hunter Dozier 139 Oliver Drake 3 Brandon Drury 120 Mauricio Dubon 28 Lucas Duda 39 Danny Duffy 2 Matt Duffy 46 Steven Duggar 73 Jon Duplantier 15 Montana DuRapau 14 Adam Duvall 41 Jarrod Dyson 130 Adam Eaton 151 Tommy Edman 92 Zach Eflin 31 Jake Elmore 20 Edwin Encarnacion 109 Adam Engel 89 Phil Ervin 94 Eduardo Escobar 158 Thairo Estrada 35 Kyle Farmer 97 Erick Fedde 20 Tim Federowicz 29 Caleb Ferguson 42 Derek Fisher 57 Jack Flaherty 35 Ryan Flaherty 14 David Fletcher 154 Chris Flexen 9 Wilmer Flores 89 Tyler Flowers 85 Mike Foltynewicz 19 Wilmer Font 14 Mike Ford 50 Logan Forsythe 101 Dexter Fowler 150 Jake Fraley 12 Ty France 69 Maikel Franco 123 Adam Frazier 152 Clint Frazier 69 Todd Frazier 133 Kyle Freeland 19 Freddie Freeman 1

<span style="color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; background-color: rgb(255, 255, 255);"><b><u>List the names and total hits for players who have scored more than 100 runs</u></b></span>

In [38]:
SELECT Name , sum(Hits) as total_hits 
FROM baseball
GROUP BY Name
HAVING SUM(Hits) > 100;


(178 row(s) affected)

Total execution time: 00:00:01.015

Name total_hits José Abreu 180 Ronald Acuna Jr. 175 Willy Adames 135 Nick Ahmed 141 Hanser Alberto 160 Ozzie Albies 189 Jorge Alfaro 113 Pete Alonso 155 Jose Altuve 149 Brian Anderson 120 Tim Anderson 167 Elvis Andrus 165 Orlando Arcia 110 Nolan Arenado 185 Luis Arraez 109 Javier Baez 149 Josh Bell 146 Cody Bellinger 170 Brandon Belt 123 Andrew Benintendi 144 Mookie Betts 176 Charlie Blackmon 182 Xander Bogaerts 190 Jackie Bradley Jr. 111 Michael Brantley 179 Ryan Braun 131 Alex Bregman 164 Kris Bryant 153 Asdrubal Cabrera 116 Melky Cabrera 106 Miguel Cabrera 139 Lorenzo Cain 146 Kole Calhoun 128 Mark Canha 112 Nicholas Castellanos 178 Harold Castro 103 Starlin Castro 172 Matt Chapman 145 Ji-Man Choi 107 Shin-Soo Choo 149 Michael Conforto 141 Garrett Cooper 107 Brandon Crawford 114 C.J. Cron 116 Nelson Cruz 141 David Dahl 113 J.D. Davis 126 Khris Davis 106 Paul DeJong 136 Ian Desmond 113 Rafael Devers 201 Josh Donaldson 142 Hunter Dozier 146 Adam Eaton 158 Edwin Encarnacion 102 Eduardo Escobar 171 David Fletcher 173 Dexter Fowler 116 Adam Frazier 154 Todd Frazier 112 Freddie Freeman 176 Freddy Galvis 145 Avisail Garcia 138 Leury Garcia 161 Brett Gardner 123 Paul Goldschmidt 155 Marwin Gonzalez 112 Niko Goodrum 105 Brian Goodwin 108 Alex Gordon 148 Dee Gordon 108 Yasmani Grandal 126 Randal Grichuk 136 Robbie Grossman 101 Vladimir Guerrero Jr. 126 Yuli Gurriel 168 Bryce Harper 149 Cesar Hernandez 171 Jason Heyward 129 Rhys Hoskins 129 Eric Hosmer 164 José Iglesias 145 Eloy Jimenez 125 Adam Jones 126 Aaron Judge 103 Howie Kendrick 115 Max Kepler 132 Kevin Kiermaier 102 Scott Kingery 118 Jason Kipnis 112 Ramon Laureano 125 DJ LeMahieu 197 Francisco Lindor 170 Evan Longoria 115 Manny Machado 150 Trey Mancini 175 Nick Markakis 118 Ketel Marte 187 Starling Marte 159 J.D. Martinez 175 Nomar Mazara 115 James McCann 120 Ryan McMahon 120 Jeff McNeil 162 Austin Meadows 154 Oscar Mercado 118 Whit Merrifield 206 Yadier Molina 113 Yoan Moncada 161 Adalberto Mondesi 109 Colin Moran 129 Mike Moustakas 133 Max Muncy 122 Daniel Murphy 122 Wil Myers 104 Omar Narvaez 119 Kevin Newman 152 Renato Nunez 132 Rougned Odor 107 Shohei Ohtani 110 Matt Olson 129 Marcell Ozuna 117 Joe Panik 107 Joc Pederson 112 David Peralta 105 Tommy Pham 155 Kevin Pillar 158 Jorge Polanco 186 Buster Posey 104 Yasiel Puig 148 Albert Pujols 120 Harold Ramirez 116 Jose Ramirez 123 Wilson Ramos 136 J.T. Realmuto 148 Josh Reddick 138 Anthony Rendon 174 Franmil Reyes 123 Bryan Reynolds 154 Anthony Rizzo 150 Victor Robles 139 Miguel Rojas 137 Amed Rosario 177 Eddie Rosario 155 Yolmer Sanchez 125 Carlos Santana 161 Danny Santana 134 Domingo Santana 114 Jonathan Schoop 111 Kyle Schwarber 132 Corey Seager 133 Jean Segura 161 Marcus Semien 187 Andrelton Simmons 105 Mallex Smith 116 Eric Sogard 115 Jorge Soler 156 Juan Soto 153 George Springer 140 Trevor Story 173 Eugenio Suarez 156 Dansby Swanson 121 Raimel Tapia 117 Fernando Tatis Jr. 106 Gleyber Torres 152 Mike Trout 137 Justin Turner 139 Trea Turner 155 Gio Urshela 139 Christian Vazquez 133 Alex Verdugo 101 Jonathan Villar 176 Luke Voit 113 Joey Votto 137 Christian Walker 137 Kolten Wong 136 Mike Yastrzemski 101 Christian Yelich 161

1. **<u>Display the player with the highest number of home runs.</u>**

In [39]:
SELECT Name, Homers
FROM baseball
WHERE Homers = (SELECT MAX(Homers) FROM baseball);


(1 row(s) affected)

Total execution time: 00:00:01.023

Name Homers Pete Alonso 53

<span style="color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; background-color: rgb(255, 255, 255);"><b><u>Show the average number of runs scored by players for each team</u></b></span>

In [40]:
SELECT Team, AVG(Runs) AS AvgRunsPerTeam
FROM baseball
GROUP BY Team;


(30 row(s) affected)

Total execution time: 00:00:01.022

Team AvgRunsPerTeam CHW 30.7826 ATL 34.2000 SFG 17.8421 TBR 27.4643 WSN 30.1034 MIN 46.8500 PIT 22.9697 MIL 24.8065 ARI 27.1000 CIN 21.2424 BAL 31.6522 MIA 19.1875 TOR 30.2500 SDP 21.9677 CLE 29.5385 CHC 27.1333 NYM 24.6250 COL 27.8333 PHI 25.7667 HOU 41.7727 OAK 33.8000 TEX 40.5000 NYY 39.2500 STL 28.2963 KCR 26.5769 LAD 26.8485 DET 24.2083 SEA 29.1154 LAA 29.5000 BOS 40.8636

1. **<u>Find the player with the maximum at-bats.</u>**

In [41]:
SELECT Name, AVG(AtBats)
from baseball
group by Name
order by AVG(AtBats) desc
LIMIT 5;

(5 row(s) affected)

Total execution time: 00:00:01.023

Name AVG(AtBats) Whit Merrifield 681.0000 Marcus Semien 657.0000 Rafael Devers 647.0000 Jonathan Villar 642.0000 Ozzie Albies 640.0000

<span style="color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; background-color: rgb(255, 255, 255);"><b><u>List the names of players with more than 30 doubles</u></b></span>

In [42]:
select Name , Doubles
from baseball
where doubles >30;


(63 row(s) affected)

Total execution time: 00:00:01.016

Name Doubles José Abreu 38 Nick Ahmed 33 Ozzie Albies 43 Brian Anderson 33 Tim Anderson 32 Nolan Arenado 31 Javier Baez 38 Josh Bell 37 Cody Bellinger 34 Brandon Belt 32 Andrew Benintendi 40 Mookie Betts 40 Charlie Blackmon 42 Xander Bogaerts 52 Michael Brantley 40 Ryan Braun 31 Alex Bregman 37 Kris Bryant 35 Nicholas Castellanos 37 Starlin Castro 31 Matt Chapman 36 Shin-Soo Choo 31 Paul DeJong 31 Ian Desmond 31 Rafael Devers 54 Josh Donaldson 33 Adam Frazier 33 Freddie Freeman 34 Alex Gordon 31 Yuli Gurriel 40 Bryce Harper 36 Cesar Hernandez 31 Rhys Hoskins 33 Max Kepler 32 Scott Kingery 34 DJ LeMahieu 33 Francisco Lindor 40 Trey Mancini 38 Ketel Marte 36 Starling Marte 31 J.D. Martinez 33 Jeff McNeil 38 Whit Merrifield 41 Yoan Moncada 34 Daniel Murphy 35 Tommy Pham 33 Kevin Pillar 37 Jorge Polanco 40 Jose Ramirez 33 J.T. Realmuto 36 Anthony Rendon 44 Bryan Reynolds 37 Victor Robles 33 Corey Seager 44 Jean Segura 37 Marcus Semien 43 Jorge Soler 33 Juan Soto 32 Trevor Story 38 Trea Turner 37 Gio Urshela 34 Jonathan Villar 33 Joey Votto 32

1. **<u>Show the player with the most triples.</u>**

In [43]:
SELECT name, triples
FROM baseball
ORDER BY triples DESC
LIMIT 1;


(1 row(s) affected)

Total execution time: 00:00:01.023

name triples Hunter Dozier 10

1. **<u>Display the names of players who have hit at least 20 home runs and have more than 80 RBIs.</u>**

In [44]:
SELECT Name , Homers,RBIs
from baseball
where Homers>=30 and RBIs > 80
limit 5;

(5 row(s) affected)

Total execution time: 00:00:01.007

Name Homers RBIs José Abreu 33 123 Ronald Acuna Jr. 41 101 Pete Alonso 53 120 Nolan Arenado 41 118 Josh Bell 37 116

<span style="color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; background-color: rgb(255, 255, 255);"><b><u>Find the player with the highest batting average (hits divided by at-bats)</u></b></span>

In [45]:
select Name , MAX(Hits/AtBats) as batingavg
from baseball
group by name
order by batingavg DESC
LIMIT 1;

(1 row(s) affected)

Total execution time: 00:00:01.031

Name batingavg R.J. Alaniz 1.0000

<span style="color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; background-color: rgb(255, 255, 255);"><b><u>List the players who have played fewer than 100 games</u></b></span>

In [46]:
select Name , Games
from baseball
WHERE Games < 100
LIMIT 10 ;

(10 row(s) affected)

Total execution time: 00:00:01.015

Name Games Cristhian Adames 10 Ehire Adrianza 83 Dario Agrazal 14 Jesus Aguilar 94 Jesus Aguilar 37 R.J. Alaniz 8 Sandy Alcantara 31 Anthony Alford 16 Austin Allen 34 Greg Allen 89

<span style="color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; background-color: rgb(255, 255, 255);"><b><u>Display the player with the maximum and minimum number of RBIs</u></b></span>

In [47]:
SELECT 
    Name, 
    max(RBIs), 
    MIN(RBIs)
FROM baseball
GROUP by Name
LIMIT  100;

(100 row(s) affected)

Total execution time: 00:00:01.023

Name max(RBIs) MIN(RBIs) José Abreu 123 123 Ronald Acuna Jr. 101 101 Cristhian Adames 2 2 Willy Adames 52 52 Matt Adams 56 56 Ehire Adrianza 22 22 Dario Agrazal 2 2 Jesus Aguilar 34 16 Nick Ahmed 82 82 R.J. Alaniz 1 1 Hanser Alberto 51 51 Ozzie Albies 86 86 Sandy Alcantara 4 4 Jorge Alfaro 57 57 Anthony Alford 1 1 Austin Allen 3 3 Greg Allen 27 27 Logan Allen 0 0 Abraham Almonte 4 4 Albert Almora 32 32 Pete Alonso 120 120 Yonder Alonso 27 10 Aaron Altherr 2 1 Jose Altuve 74 74 Yordan Álvarez 78 78 Brett Anderson 0 0 Brian Anderson 66 66 Chase Anderson 1 1 Shaun Anderson 0 0 Tim Anderson 56 56 Tyler Anderson 0 0 Elvis Andrus 72 72 Miguel Andujar 1 1 Aristides Aquino 47 47 Chris Archer 2 2 Orlando Arcia 59 59 Nolan Arenado 118 118 Randy Arozarena 2 2 Luis Arraez 28 28 Jake Arrieta 4 4 Christian Arroyo 7 7 Humberto Arteaga 4 4 Willians Astudillo 21 21 Tyler Austin 20 0 Abiatal Avelino 1 1 Alex Avila 24 24 Harrison Bader 39 39 Javier Baez 85 85 Homer Bailey 1 1 Austin Barnes 25 25 Tucker Barnhart 40 40 Steven Baron 1 1 Franklin Barreto 5 5 Trevor Bauer 0 0 Jake Bauers 43 43 Matt Beaty 46 46 Gordon Beckham 15 15 Tim Beckham 47 47 Tyler Beede 1 1 Josh Bell 116 116 Cody Bellinger 115 115 Brandon Belt 57 57 Anthony Bemboom 3 1 Andrew Benintendi 68 68 Jon Berti 24 24 Mookie Betts 80 80 Bo Bichette 21 21 Cavan Biggio 48 48 Greg Bird 1 1 Braden Bishop 4 4 Charlie Blackmon 86 86 Alex Blandino 3 3 Xander Bogaerts 117 117 Skye Bolt 0 0 Jorge Bonifacio 3 3 David Bote 41 41 Justin Bour 26 26 Peter Bourjos 2 2 Matthew Boyd 0 0 Brad Brach 1 1 Bobby Bradley 4 4 Jackie Bradley Jr. 62 62 Michael Brantley 90 90 Steven Brault 2 2 Ryan Braun 75 75 John Brebbia 0 0 Alex Bregman 112 112 Lewis Brinson 15 15 Socrates Brito 2 2 Mike Brosseau 16 16 Seth Brown 13 13 Keon Broxton 9 2 Jay Bruce 31 28 Kris Bryant 77 77 Walker Buehler 2 2 Madison Bumgarner 4 4 Corbin Burnes 0 0 Drew Butera 3 3 Byron Buxton 46 46 Asdrubal Cabrera 51 40

<span style="color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; background-color: rgb(255, 255, 255);"><b><u>Show the names of players who have played exactly 150 games</u></b></span>

In [48]:
select Name , Games
FROM baseball
WHERE Games=100;

(5 row(s) affected)

Total execution time: 00:00:01.011

Name Games Cavan Biggio 100 Nicholas Castellanos 100 David Dahl 100 Michael Lorenzen 100 Yasiel Puig 100

<span style="color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; background-color: rgb(255, 255, 255);"><b><u>Find the player with the highest number of runs per game (runs divided by games)</u></b></span>

In [49]:
SELECT Name , Runs/Games as AvgRunsPergame
from baseball
ORDER by AvgRunsPergame DESC
limit 10;


(10 row(s) affected)

Total execution time: 00:00:01.031

Name AvgRunsPergame Trent Thornton 2.0000 Mookie Betts 0.9000 Nicholas Castellanos 0.8431 Rafael Devers 0.8269 Mike Trout 0.8209 Ronald Acuna Jr. 0.8141 Anthony Rendon 0.8014 Charlie Blackmon 0.8000 Ryan Lavarnway 0.8000 George Springer 0.7869

1. **<u>List the players who have hit at least 10 home runs and played for 'NYY' or 'BOS'.</u>**

In [50]:
SELECT Name , Homers, Team
FROM baseball
where Homers >=10 and Team = 'BOS' OR TEAM = 'NYY'

(33 row(s) affected)

Total execution time: 00:00:01.103

Name Homers Team Miguel Andujar 0 NYY Andrew Benintendi 13 BOS Mookie Betts 29 BOS Greg Bird 1 NYY Xander Bogaerts 33 BOS Jackie Bradley Jr. 21 BOS Michael Chavis 18 BOS Rafael Devers 32 BOS Edwin Encarnacion 13 NYY Thairo Estrada 3 NYY Mike Ford 12 NYY Clint Frazier 12 NYY Brett Gardner 28 NYY Didi Gregorius 16 NYY Aaron Hicks 12 NYY Kyle Higashioka 3 NYY Aaron Judge 27 NYY DJ LeMahieu 26 NYY J.D. Martinez 36 BOS Cameron Maybin 11 NYY Kendrys Morales 1 NYY Mitch Moreland 19 BOS Austin Romine 8 NYY Gary Sanchez 34 NYY Giancarlo Stanton 3 NYY Mike Tauchman 13 NYY Gleyber Torres 38 NYY Troy Tulowitzki 1 NYY Gio Urshela 21 NYY Breyvic Valera 0 NYY Christian Vazquez 23 BOS Luke Voit 21 NYY Tyler Wade 2 NYY

1. **<u>Display the player with the highest total runs and the lowest total hits.</u>**

In [51]:
SELECT Name, MAX(Runs),MIN(Runs)
from baseball
GROUP by Name;


(768 row(s) affected)

Total execution time: 00:00:01.023

Name MAX(Runs) MIN(Runs) José Abreu 85 85 Ronald Acuna Jr. 127 127 Cristhian Adames 1 1 Willy Adames 69 69 Matt Adams 42 42 Ehire Adrianza 34 34 Dario Agrazal 0 0 Jesus Aguilar 26 13 Nick Ahmed 79 79 R.J. Alaniz 0 0 Hanser Alberto 62 62 Ozzie Albies 102 102 Sandy Alcantara 2 2 Jorge Alfaro 44 44 Anthony Alford 3 3 Austin Allen 4 4 Greg Allen 30 30 Logan Allen 1 1 Abraham Almonte 11 11 Albert Almora 41 41 Pete Alonso 103 103 Yonder Alonso 23 11 Aaron Altherr 6 2 Jose Altuve 89 89 Yordan Álvarez 58 58 Brett Anderson 1 1 Brian Anderson 57 57 Chase Anderson 1 1 Shaun Anderson 0 0 Tim Anderson 81 81 Tyler Anderson 1 1 Elvis Andrus 81 81 Miguel Andujar 1 1 Aristides Aquino 31 31 Chris Archer 1 1 Orlando Arcia 51 51 Nolan Arenado 102 102 Randy Arozarena 4 4 Luis Arraez 54 54 Jake Arrieta 2 2 Christian Arroyo 8 8 Humberto Arteaga 11 11 Willians Astudillo 28 28 Tyler Austin 24 1 Abiatal Avelino 0 0 Alex Avila 22 22 Harrison Bader 54 54 Javier Baez 89 89 Homer Bailey 0 0 Austin Barnes 28 28 Tucker Barnhart 32 32 Steven Baron 0 0 Franklin Barreto 6 6 Trevor Bauer 1 0 Jake Bauers 46 46 Matt Beaty 36 36 Gordon Beckham 29 29 Tim Beckham 39 39 Tyler Beede 2 2 Josh Bell 94 94 Cody Bellinger 121 121 Brandon Belt 76 76 Anthony Bemboom 2 0 Andrew Benintendi 72 72 Jon Berti 52 52 Mookie Betts 135 135 Bo Bichette 32 32 Cavan Biggio 66 66 Greg Bird 6 6 Braden Bishop 3 3 Charlie Blackmon 112 112 Alex Blandino 6 6 Xander Bogaerts 110 110 Skye Bolt 1 1 Jorge Bonifacio 3 3 David Bote 47 47 Justin Bour 18 18 Peter Bourjos 4 4 Matthew Boyd 0 0 Brad Brach 0 0 Bobby Bradley 4 4 Jackie Bradley Jr. 69 69 Michael Brantley 88 88 Steven Brault 8 8 Ryan Braun 70 70 John Brebbia 1 1 Alex Bregman 122 122 Lewis Brinson 15 15 Socrates Brito 5 5 Mike Brosseau 17 17 Seth Brown 11 11 Keon Broxton 14 5 Jay Bruce 27 16 Kris Bryant 108 108 Walker Buehler 5 5 Madison Bumgarner 4 4 Corbin Burnes 0 0 Drew Butera 6 6 Byron Buxton 48 48 Asdrubal Cabrera 45 24 Melky Cabrera 43 43 Miguel Cabrera 41 41 Trevor Cahill 0 0 Lorenzo Cain 75 75 Kole Calhoun 92 92 Willie Calhoun 51 51 Johan Camargo 31 31 Jeimer Candelario 33 33 Mark Canha 80 80 Robinson Cano 46 46 Victor Caratini 31 31 Matt Carpenter 59 59 Curt Casali 24 24 Nicholas Castellanos 57 43 Luis Castillo 3 3 Welington Castillo 19 19 Wilkin Castillo 0 0 Harold Castro 30 30 Jason Castro 39 39 Starlin Castro 68 68 Willi Castro 10 10 Jake Cave 28 28 Juan Centeno 0 0 Francisco Cervelli 11 4 Jhoulys Chacin 2 2 Yu Chang 8 8 Matt Chapman 102 102 Tyler Chatwood 1 1 Jesse Chavez 0 0 Michael Chavis 46 46 Robinson Chirinos 57 57 Ji-Man Choi 54 54 Shin-Soo Choo 93 93 Nick Ciuffo 0 0 Taylor Clarke 2 2 Zack Collins 10 10 Christian Colon 1 1 Michael Conforto 90 90 Willson Contreras 57 57 Garrett Cooper 52 52 Patrick Corbin 3 3 Ryan Cordell 22 22 Franchy Cordero 2 2 Carlos Correa 42 42 Ryan Court 1 1 Kaleb Cowart 1 1 Zack Cozart 4 4 Brandon Crawford 58 58 J.P. Crawford 43 43 C.J. Cron 51 51 Kevin Cron 12 12 Nelson Cruz 81 81 Charlie Culberson 14 14 Cheslor Cuthbert 24 24 Travis d'Arnaud 50 2 David Dahl 67 67 Yu Darvish 1 1 Zach Davies 3 3 Chris Davis 26 26 J.D. Davis 65 65 Jaylin Davis 2 2 Johnny Davis 5 5 Jonathan Davis 8 8 Khris Davis 61 61 Rajai Davis 4 4 Taylor Davis 2 2 Yonathan Daza 7 7 Austin Dean 17 17 Jacob deGrom 4 4 Paul DeJong 97 97 Nicky Delmonico 6 6 Travis Demeritte 24 24 Daniel Descalso 20 20 Anthony DeSclafani 0 0 Delino DeShields 42 42 Ian Desmond 64 64 Rafael Devers 129 129 Aledmys Díaz 36 36 Elias Diaz 31 31 Isan Diaz 17 17 Yandy Díaz 53 53 Alex Dickerson 28 1 Corey Dickerson 20 13 Derek Dietrich 41 41 Wilmer Difo 15 15 Nick Dini 11 11 Brandon Dixon 41 41 Josh Donaldson 96 96 Brian Dozier 54 54 Hunter Dozier 75 75 Oliver Drake 0 0 Brandon Drury 43 43 Mauricio Dubon 12 12 Lucas Duda 7 7 Danny Duffy 0 0 Matt Duffy 12 12 Steven Duggar 26 26 Jon Duplantier 1 1 Montana DuRapau 0 0 Adam Duvall 17 17 Jarrod Dyson 65 65 Adam Eaton 103 103 Tommy Edman 59 59 Zach Eflin 3 3 Jake Elmore 3 3 Edwin Encarnacion 48 33 Adam Engel 26 26 Phil

1. **<u>Show the players with more than 10 doubles and a batting average greater than 0.300.......?</u>**

In [54]:
SELECT Name , Doubles ,AVG(AtBats) as avg_run
from baseball
GROUP by Name ,GROUP BY Doubles
having Doubles >=10;

Total execution time: 00:00:00.007

: 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'GROUP BY Doubles
having Doubles >=10' at line 3

<span style="color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; background-color: rgb(255, 255, 255);"><b><u>List the players who have hit a home run but have less than 50 runs scored</u></b></span>

In [55]:
select Name ,Runs,Hits
from baseball
WHERE Hits >=1 and Runs <50;

(628 row(s) affected)

Total execution time: 00:00:01.025

Name Runs Hits Cristhian Adames 1 7 Matt Adams 42 70 Ehire Adrianza 34 55 Dario Agrazal 0 2 Jesus Aguilar 26 50 Jesus Aguilar 13 24 R.J. Alaniz 0 1 Sandy Alcantara 2 5 Jorge Alfaro 44 113 Anthony Alford 3 5 Austin Allen 4 14 Greg Allen 30 53 Logan Allen 1 1 Abraham Almonte 11 9 Albert Almora 41 80 Yonder Alonso 23 39 Yonder Alonso 11 19 Aaron Altherr 2 1 Aaron Altherr 6 4 Brett Anderson 1 4 Chase Anderson 1 4 Shaun Anderson 0 2 Tyler Anderson 1 1 Miguel Andujar 1 6 Aristides Aquino 31 53 Chris Archer 1 4 Randy Arozarena 4 6 Jake Arrieta 2 7 Christian Arroyo 8 11 Humberto Arteaga 11 24 Willians Astudillo 28 51 Tyler Austin 1 1 Tyler Austin 24 24 Tyler Austin 5 4 Abiatal Avelino 0 2 Alex Avila 22 34 Homer Bailey 0 2 Austin Barnes 28 43 Tucker Barnhart 32 73 Steven Baron 0 2 Franklin Barreto 6 7 Trevor Bauer 0 1 Trevor Bauer 1 2 Jake Bauers 46 84 Matt Beaty 36 66 Gordon Beckham 29 48 Tim Beckham 39 72 Tyler Beede 2 4 Anthony Bemboom 0 2 Anthony Bemboom 2 5 Bo Bichette 32 61 Greg Bird 6 6 Braden Bishop 3 6 Alex Blandino 6 9 Skye Bolt 1 1 Jorge Bonifacio 3 7 David Bote 47 78 Justin Bour 18 26 Peter Bourjos 4 4 Matthew Boyd 0 1 Brad Brach 0 1 Bobby Bradley 4 8 Steven Brault 8 14 John Brebbia 1 1 Lewis Brinson 15 39 Socrates Brito 5 3 Mike Brosseau 17 36 Seth Brown 11 22 Keon Broxton 14 21 Keon Broxton 5 6 Keon Broxton 5 7 Jay Bruce 27 35 Jay Bruce 16 32 Walker Buehler 5 5 Madison Bumgarner 4 8 Corbin Burnes 0 2 Drew Butera 6 7 Byron Buxton 48 71 Asdrubal Cabrera 45 76 Asdrubal Cabrera 24 40 Melky Cabrera 43 106 Miguel Cabrera 41 139 Trevor Cahill 0 1 Johan Camargo 31 54 Jeimer Candelario 33 68 Robinson Cano 46 100 Victor Caratini 31 65 Curt Casali 24 52 Nicholas Castellanos 43 68 Luis Castillo 3 6 Welington Castillo 19 48 Wilkin Castillo 0 1 Harold Castro 30 103 Jason Castro 39 55 Willi Castro 10 23 Jake Cave 28 51 Juan Centeno 0 2 Francisco Cervelli 11 21 Francisco Cervelli 4 9 Jhoulys Chacin 2 3 Yu Chang 8 13 Tyler Chatwood 1 3 Jesse Chavez 0 1 Michael Chavis 46 88 Nick Ciuffo 0 1 Taylor Clarke 2 6 Zack Collins 10 16 Christian Colon 1 3 Patrick Corbin 3 6 Ryan Cordell 22 48 Franchy Cordero 2 5 Carlos Correa 42 78 Ryan Court 1 5 Kaleb Cowart 1 4 Zack Cozart 4 12 J.P. Crawford 43 78 Kevin Cron 12 15 Charlie Culberson 14 35 Cheslor Cuthbert 24 76 Travis d'Arnaud 2 2 Yu Darvish 1 5 Zach Davies 3 8 Chris Davis 26 55 Jaylin Davis 2 7 Johnny Davis 5 1 Jonathan Davis 8 15 Rajai Davis 4 5 Taylor Davis 2 3 Yonathan Daza 7 20 Austin Dean 17 40 Jacob deGrom 4 13 Nicky Delmonico 6 13 Travis Demeritte 24 38 Daniel Descalso 20 29 Anthony DeSclafani 0 6 Delino DeShields 42 89 Aledmys Díaz 36 57 Elias Diaz 31 73 Isan Diaz 17 31 Alex Dickerson 1 3 Alex Dickerson 28 45 Corey Dickerson 20 40 Corey Dickerson 13 39 Derek Dietrich 41 47 Wilmer Difo 15 33 Nick Dini 11 11 Brandon Dixon 41 97 Oliver Drake 0 1 Brandon Drury 43 91 Mauricio Dubon 12 29 Lucas Duda 7 18 Danny Duffy 0 1 Matt Duffy 12 37 Steven Duggar 26 61 Jon Duplantier 1 1 Montana DuRapau 0 1 Adam Duvall 17 32 Zach Eflin 3 8 Jake Elmore 3 10 Edwin Encarnacion 48 58 Edwin Encarnacion 33 44 Adam Engel 26 55 Phil Ervin 30 64 Thairo Estrada 12 16 Kyle Farmer 22 42 Erick Fedde 1 2 Tim Federowicz 6 12 Caleb Ferguson 1 1 Derek Fisher 9 12 Derek Fisher 14 15 Jack Flaherty 4 10 Ryan Flaherty 4 3 Chris Flexen 0 1 Wilmer Flores 31 84 Tyler Flowers 36 62 Mike Foltynewicz 1 2 Wilmer Font 0 1 Mike Ford 30 37 Logan Forsythe 38 72 Jake Fraley 3 6 Ty France 20 43 Maikel Franco 48 91 Clint Frazier 31 60 Kyle Freeland 1 2 Mike Freeman 27 49 David Freese 35 51 David Freitas 1 1 Max Fried 11 11 Josh Fuentes 8 12 Drew Gagnon 0 1 Rocky Gale 1 2 Cam Gallagher 14 30 Zac Gallen 1 1 Zac Gallen 1 1 Isaac Galloway 6 9 Freddy Galvis 12 25 Ben Gamel 47 77 Aramis Garcia 5 6 Robel Garcia 8 15 Kyle Garlick 8 12 Dustin Garneau 11 16 Dustin Garneau 3 5 Kevin Gausman 1 1 Scooter Gennett 4 15 Scooter Gennett 11 15 Mike Gerber 0 1 Lucas Giolito 0 1 Zack Godley 3 3 Ryan Goins 13 36 Yan Gomes 36 70 Carlos Gomez 10 17 Tony Gonsolin 2 

1. **<u>Display the names of players who have at least 40 hits and have played fewer than 50 games.</u>**

In [4]:
select Name , Hits ,Games
from baseball
having Hits >=140 
LIMIT  5;

(5 row(s) affected)

Total execution time: 00:00:01.015

Name Hits Games José Abreu 180 159 Ronald Acuna Jr. 175 156 Nick Ahmed 141 158 Hanser Alberto 160 139 Ozzie Albies 189 160

<span style="color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; background-color: rgb(255, 255, 255);"><b><u>Find the player with the highest batting average among those with at least 300 at-bats</u></b></span>

In [67]:
SELECT Name, AVG(AtBats) AS avg_batting
FROM baseball
WHERE AtBats >= 300
GROUP BY Name
limit 10;


(10 row(s) affected)

Total execution time: 00:00:01.023

Name avg_batting José Abreu 634.0000 Ronald Acuna Jr. 626.0000 Willy Adames 531.0000 Matt Adams 310.0000 Nick Ahmed 556.0000 Hanser Alberto 524.0000 Ozzie Albies 640.0000 Jorge Alfaro 431.0000 Albert Almora 339.0000 Pete Alonso 597.0000

1. **<u>List the players with more than 25 doubles and an average of at least 3 RBIs per game.</u>**

In [72]:
SELECT Name, SUM(Doubles) AS Doubles, AVG(RBIs) AS avgrb
FROM baseball
GROUP BY Name
HAVING SUM(Doubles) > 25 AND AVG(RBIs) >= 3;


(111 row(s) affected)

Total execution time: 00:00:01.015

Name Doubles avgrb José Abreu 38 123.0000 Nick Ahmed 33 82.0000 Ozzie Albies 43 86.0000 Pete Alonso 30 120.0000 Jose Altuve 27 74.0000 Yordan Álvarez 26 78.0000 Brian Anderson 33 66.0000 Tim Anderson 32 56.0000 Elvis Andrus 27 72.0000 Nolan Arenado 31 118.0000 Javier Baez 38 85.0000 Josh Bell 37 116.0000 Cody Bellinger 34 115.0000 Brandon Belt 32 57.0000 Andrew Benintendi 40 68.0000 Mookie Betts 40 80.0000 Charlie Blackmon 42 86.0000 Xander Bogaerts 52 117.0000 Jackie Bradley Jr. 28 62.0000 Michael Brantley 40 90.0000 Ryan Braun 31 75.0000 Alex Bregman 37 112.0000 Kris Bryant 35 77.0000 Byron Buxton 30 46.0000 Lorenzo Cain 30 48.0000 Kole Calhoun 29 74.0000 Robinson Cano 28 39.0000 Nicholas Castellanos 58 36.5000 Starlin Castro 31 86.0000 Matt Chapman 36 91.0000 Shin-Soo Choo 31 61.0000 Michael Conforto 29 92.0000 Nelson Cruz 26 108.0000 David Dahl 28 61.0000 Paul DeJong 31 78.0000 Ian Desmond 31 65.0000 Rafael Devers 54 115.0000 Corey Dickerson 28 29.5000 Josh Donaldson 33 94.0000 Hunter Dozier 29 84.0000 Eduardo Escobar 29 118.0000 David Fletcher 30 49.0000 Adam Frazier 33 50.0000 Freddie Freeman 34 121.0000 Freddy Galvis 28 35.0000 Leury Garcia 27 40.0000 Brett Gardner 26 74.0000 Niko Goodrum 27 45.0000 Brian Goodwin 29 47.0000 Alex Gordon 31 76.0000 Yasmani Grandal 26 77.0000 Randal Grichuk 29 80.0000 Vladimir Guerrero Jr. 26 69.0000 Yuli Gurriel 40 104.0000 Bryce Harper 36 114.0000 Cesar Hernandez 31 71.0000 Rhys Hoskins 33 85.0000 Eric Hosmer 29 99.0000 Max Kepler 32 90.0000 Scott Kingery 34 55.0000 Ramon Laureano 29 67.0000 DJ LeMahieu 33 102.0000 Francisco Lindor 40 74.0000 Trey Mancini 38 97.0000 Ketel Marte 36 92.0000 Starling Marte 31 82.0000 J.D. Martinez 33 105.0000 Nomar Mazara 27 66.0000 James McCann 26 60.0000 Jeff McNeil 38 75.0000 Austin Meadows 29 89.0000 Whit Merrifield 41 74.0000 Yoan Moncada 34 79.0000 Colin Moran 30 80.0000 Mike Moustakas 30 87.0000 Daniel Murphy 35 78.0000 Rougned Odor 30 93.0000 Matt Olson 26 91.0000 David Peralta 29 57.0000 Tommy Pham 33 68.0000 Kevin Pillar 37 44.0000 Jorge Polanco 40 79.0000 Yasiel Puig 30 42.0000 Jose Ramirez 33 83.0000 J.T. Realmuto 36 83.0000 Anthony Rendon 44 126.0000 Bryan Reynolds 37 68.0000 Anthony Rizzo 29 94.0000 Victor Robles 33 65.0000 Miguel Rojas 29 46.0000 Amed Rosario 30 72.0000 Eddie Rosario 28 109.0000 Carlos Santana 30 93.0000 Kyle Schwarber 29 92.0000 Corey Seager 44 87.0000 Jean Segura 37 60.0000 Marcus Semien 43 92.0000 Jorge Soler 33 117.0000 Juan Soto 32 110.0000 Trevor Story 38 85.0000 Dansby Swanson 26 65.0000 Chris Taylor 29 52.0000 Gleyber Torres 26 90.0000 Mike Trout 27 104.0000 Trea Turner 37 57.0000 Gio Urshela 34 74.0000 Christian Vazquez 26 72.0000 Jonathan Villar 33 73.0000 Joey Votto 32 47.0000 Christian Walker 26 73.0000 Christian Yelich 29 97.0000

In [73]:

select Name ,Doubles , AVG(RBIs) as avgrb
from baseball
group by Name , Doubles
having Doubles >25 and avgrb >=3

(108 row(s) affected)

Total execution time: 00:00:01.063

Name Doubles avgrb José Abreu 38 123.0000 Nick Ahmed 33 82.0000 Ozzie Albies 43 86.0000 Pete Alonso 30 120.0000 Jose Altuve 27 74.0000 Yordan Álvarez 26 78.0000 Brian Anderson 33 66.0000 Tim Anderson 32 56.0000 Elvis Andrus 27 72.0000 Nolan Arenado 31 118.0000 Javier Baez 38 85.0000 Josh Bell 37 116.0000 Cody Bellinger 34 115.0000 Brandon Belt 32 57.0000 Andrew Benintendi 40 68.0000 Mookie Betts 40 80.0000 Charlie Blackmon 42 86.0000 Xander Bogaerts 52 117.0000 Jackie Bradley Jr. 28 62.0000 Michael Brantley 40 90.0000 Ryan Braun 31 75.0000 Alex Bregman 37 112.0000 Kris Bryant 35 77.0000 Byron Buxton 30 46.0000 Lorenzo Cain 30 48.0000 Kole Calhoun 29 74.0000 Robinson Cano 28 39.0000 Nicholas Castellanos 37 37.0000 Starlin Castro 31 86.0000 Matt Chapman 36 91.0000 Shin-Soo Choo 31 61.0000 Michael Conforto 29 92.0000 Nelson Cruz 26 108.0000 David Dahl 28 61.0000 Paul DeJong 31 78.0000 Ian Desmond 31 65.0000 Rafael Devers 54 115.0000 Josh Donaldson 33 94.0000 Hunter Dozier 29 84.0000 Eduardo Escobar 29 118.0000 David Fletcher 30 49.0000 Adam Frazier 33 50.0000 Freddie Freeman 34 121.0000 Leury Garcia 27 40.0000 Brett Gardner 26 74.0000 Niko Goodrum 27 45.0000 Brian Goodwin 29 47.0000 Alex Gordon 31 76.0000 Yasmani Grandal 26 77.0000 Randal Grichuk 29 80.0000 Vladimir Guerrero Jr. 26 69.0000 Yuli Gurriel 40 104.0000 Bryce Harper 36 114.0000 Cesar Hernandez 31 71.0000 Rhys Hoskins 33 85.0000 Eric Hosmer 29 99.0000 Max Kepler 32 90.0000 Scott Kingery 34 55.0000 Ramon Laureano 29 67.0000 DJ LeMahieu 33 102.0000 Francisco Lindor 40 74.0000 Trey Mancini 38 97.0000 Ketel Marte 36 92.0000 Starling Marte 31 82.0000 J.D. Martinez 33 105.0000 Nomar Mazara 27 66.0000 James McCann 26 60.0000 Jeff McNeil 38 75.0000 Austin Meadows 29 89.0000 Whit Merrifield 41 74.0000 Yoan Moncada 34 79.0000 Colin Moran 30 80.0000 Mike Moustakas 30 87.0000 Daniel Murphy 35 78.0000 Rougned Odor 30 93.0000 Matt Olson 26 91.0000 David Peralta 29 57.0000 Tommy Pham 33 68.0000 Kevin Pillar 37 87.0000 Jorge Polanco 40 79.0000 Jose Ramirez 33 83.0000 J.T. Realmuto 36 83.0000 Anthony Rendon 44 126.0000 Bryan Reynolds 37 68.0000 Anthony Rizzo 29 94.0000 Victor Robles 33 65.0000 Miguel Rojas 29 46.0000 Amed Rosario 30 72.0000 Eddie Rosario 28 109.0000 Carlos Santana 30 93.0000 Kyle Schwarber 29 92.0000 Corey Seager 44 87.0000 Jean Segura 37 60.0000 Marcus Semien 43 92.0000 Jorge Soler 33 117.0000 Juan Soto 32 110.0000 Trevor Story 38 85.0000 Dansby Swanson 26 65.0000 Chris Taylor 29 52.0000 Gleyber Torres 26 90.0000 Mike Trout 27 104.0000 Trea Turner 37 57.0000 Gio Urshela 34 74.0000 Christian Vazquez 26 72.0000 Jonathan Villar 33 73.0000 Joey Votto 32 47.0000 Christian Walker 26 73.0000 Christian Yelich 29 97.0000